entities/storobj/storage_object_test.go

```diff
@@ -239,6 +239,46 @@ func TestStorageObjectUnMarshallingMultiVector(t *testing.T) {
 			assert.ElementsMatch(t, after.MultiVectors["vector4"], before.MultiVectors["vector4"])
 			assert.ElementsMatch(t, after.MultiVectors["vector5"], before.MultiVectors["vector5"])
 		})
+
+		t.Run("check multi vectors optional", func(t *testing.T) {
+			t.Run("empty additional, empty multi vectors", func(t *testing.T) {
+				afterMultiVectorsOptional, err := FromBinaryOptional(asBinary, additional.Properties{}, nil)
+				require.Nil(t, err)
+				require.Nil(t, afterMultiVectorsOptional.MultiVectors)
+			})
+
+			t.Run("multi vector in additional via named vectors", func(t *testing.T) {
+				afterMultiVectorsOptional, err := FromBinaryOptional(asBinary, additional.Properties{
+					Vectors: []string{"vector4"},
+				}, nil)
+				require.Nil(t, err)
+				require.NotEmpty(t, afterMultiVectorsOptional.MultiVectors)
+				require.Len(t, afterMultiVectorsOptional.MultiVectors, 1)
+				require.Equal(t, before.MultiVectors["vector4"], afterMultiVectorsOptional.MultiVectors["vector4"])
+			})
+
+			t.Run("named vector and multi vector in additional", func(t *testing.T) {
+				afterMultiVectorsOptional, err := FromBinaryOptional(asBinary, additional.Properties{
+					Vectors: []string{"vector2", "vector4"},
+				}, nil)
+				require.Nil(t, err)
+				require.NotEmpty(t, afterMultiVectorsOptional.Vectors)
+				require.NotEmpty(t, afterMultiVectorsOptional.MultiVectors)
+				// TODO should this length be 1 or 2?
+				require.Len(t, afterMultiVectorsOptional.Vectors, 2)
+				require.Len(t, afterMultiVectorsOptional.MultiVectors, 1)
+				// TODO this passes, should we modify FromBinaryOptional to only populate vector2?
+				require.Equal(t, before.Vectors["vector1"], afterMultiVectorsOptional.Vectors["vector1"])
+				require.Equal(t, before.Vectors["vector2"], afterMultiVectorsOptional.Vectors["vector2"])
+				// TODO why doesn't FromBinaryOptional set .Object.Vectors?
+				// require.Equal(t, before.Vectors["vector1"], afterMultiVectorsOptional.Object.Vectors["vector1"])
+				// require.Equal(t, before.Vectors["vector2"], afterMultiVectorsOptional.Object.Vectors["vector2"])
+				require.Equal(t, before.MultiVectors["vector4"], afterMultiVectorsOptional.MultiVectors["vector4"])
+				// TODO do we need to populate object.Vectors with multivectors? or should we add a new MultiVectors field?
+				// require.Equal(t, before.MultiVectors["vector4"], afterMultiVectorsOptional.Object.Vectors["vector4"])
```

Comment:
Should we populate `afterMultiVectorsOptional.Object.Vectors` with `vector2` and `vector4` here? I'm not familiar with how that field is used